control-plane/agents/src/bin/core/controller/resources/operations_helper.rs

```diff
@@ -937,3 +942,23 @@ impl ResourceSpecsLocked {
         Ok(())
     }
 }
+
+/// Helper function to extract the volume groups from volumes on startup.
+fn get_volume_group_specs(volume_specs: &[VolumeSpec]) -> Vec<VolumeGroupSpec> {
+    let mut volume_group_specs: Vec<VolumeGroupSpec> = Vec::new();
+    for volume_spec in volume_specs.iter() {
+        if let Some(volume_group) = &volume_spec.volume_group {
+            let volume_group_spec =
+                VolumeGroupSpec::new(volume_group.name(), vec![volume_spec.uuid.clone()]);
+            let existing_index = volume_group_specs
+                .iter()
+                .position(|x| x.name() == volume_group_spec.name());
+            if let Some(index) = existing_index {
```

Comment:
why not simply use find_mut ?

Reply:
Good point.

Reply:
Done. But time complexity is same though.

Reply:
Not for performance, just to avoid avoid to index since we can get the object straight away.

Reply:
Ohh, Right.